gc/gc-repository-jdbc/src/main/java/org/projectnessie/gc/contents/jdbc/JdbcPersistenceSpi.java

```diff
@@ -87,6 +88,15 @@ public interface Builder {
     JdbcPersistenceSpi build();
   }
 
+  @Value.Lazy
+  protected String productName() {
+    try (Connection conn = dataSource().getConnection()) {
```

Comment:
Not strongly opposed to get the "product name" this way - although using a JDBC connection during an "immutables.build()" isn't "nice".
Isn't there a way to get that information from the data-source directly?

Reply:
It's marked lazy exactly because of that – the JDBC connection will be created only when the method is called from `decorateInsertStatement`.

Reply:
Unfortunately there is nothing useful under `((AgroalDataSource) dataSource()).getConfiguration()` that could hint at which database is being used.

We _could_ parse the user-provided JDBC URL; but then we'd need to pass it around. Not sure that's better.